### ISSUE TITLE
Implicit conversion definition on MacOS

### ISSUE THREAD

**Issue by jamesmunns** [DavidBuchanan314/dag-cbrrr#4]
I get the following warnings and errors on MacOS Sonoma 14.3:

```
pip3 install .
Processing /Users/james/personal/satur-dag/vendor/dag-cbrrr
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: cbrrr
  Building wheel for cbrrr (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for cbrrr (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [59 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build/lib.macosx-13-arm64-cpython-311
      creating build/lib.macosx-13-arm64-cpython-311/cbrrr
      copying src/cbrrr/__init__.py -> build/lib.macosx-13-arm64-cpython-311/cbrrr
      running egg_info
      writing cbrrr.egg-info/PKG-INFO
      writing dependency_links to cbrrr.egg-info/dependency_links.txt
      writing requirements to cbrrr.egg-info/requires.txt
      writing top-level names to cbrrr.egg-info/top_level.txt
      reading manifest file 'cbrrr.egg-info/SOURCES.txt'
      adding license file 'LICENSE'
      writing manifest file 'cbrrr.egg-info/SOURCES.txt'
      copying src/cbrrr/_cbrrr.c -> build/lib.macosx-13-arm64-cpython-311/cbrrr
      running build_ext
      building '_cbrrr' extension
      clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -I/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c src/cbrrr/_cbrrr.c -o build/temp.macosx-13-arm64-cpython-311/src/cbrrr/_cbrrr.o -O3 -Wall -Wextra -Wpedantic -std=c99
      src/cbrrr/_cbrrr.c:17:28: warning: '$' in identifier [-Wdollar-in-identifier-extension]
      static PyObject *PY_STRING_$LINK;
                                 ^
      src/cbrrr/_cbrrr.c:18:28: warning: '$' in identifier [-Wdollar-in-identifier-extension]
      static PyObject *PY_STRING_$BYTES;
                                 ^
      src/cbrrr/_cbrrr.c:260:55: warning: incompatible pointer types passing 'size_t *' (aka 'unsigned long *') to parameter of type 'uint64_t *' (aka 'unsigned long long *') [-Wincompatible-pointer-types]
              res = cbrrr_parse_minimal_varint(&buf[idx], len-idx, str_len);
                                                                   ^~~~~~~
      src/cbrrr/_cbrrr.c:183:70: note: passing argument to parameter 'value' here
      cbrrr_parse_minimal_varint(const uint8_t *buf, size_t len, uint64_t *value)
                                                                           ^
      src/cbrrr/_cbrrr.c:376:47: warning: '$' in identifier [-Wdollar-in-identifier-extension]
                              if (PyDict_SetItem(token->value, PY_STRING_$BYTES, tmp) != 0) {
                                                                         ^
      src/cbrrr/_cbrrr.c:449:47: warning: '$' in identifier [-Wdollar-in-identifier-extension]
                              if (PyDict_SetItem(token->value, PY_STRING_$LINK, tmp) != 0) {
                                                                         ^
      src/cbrrr/_cbrrr.c:1312:23: error: call to undeclared function 'htobe64'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                              uint64_t dub_int = htobe64(((union {uint64_t num; double dub;}){.dub=doubleval}).num);
                                                 ^
      src/cbrrr/_cbrrr.c:1411:12: warning: '$' in identifier [-Wdollar-in-identifier-extension]
              PY_STRING_$LINK = PyUnicode_InternFromString("$link");
                        ^
      src/cbrrr/_cbrrr.c:1412:12: warning: '$' in identifier [-Wdollar-in-identifier-extension]
              PY_STRING_$BYTES = PyUnicode_InternFromString("$bytes");
                        ^
      src/cbrrr/_cbrrr.c:1419:16: warning: '$' in identifier [-Wdollar-in-identifier-extension]
                      || PY_STRING_$LINK == NULL
                                   ^
      src/cbrrr/_cbrrr.c:1420:16: warning: '$' in identifier [-Wdollar-in-identifier-extension]
                      || PY_STRING_$BYTES == NULL
                                   ^
      src/cbrrr/_cbrrr.c:1427:24: warning: '$' in identifier [-Wdollar-in-identifier-extension]
                      Py_XDECREF(PY_STRING_$LINK);
                                           ^
      src/cbrrr/_cbrrr.c:1428:24: warning: '$' in identifier [-Wdollar-in-identifier-extension]
                      Py_XDECREF(PY_STRING_$BYTES);
                                           ^
      11 warnings and 1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cbrrr
Failed to build cbrrr
ERROR: Could not build wheels for cbrrr, which is required to install pyproject.toml-based projects
```

It seems there is a missing include of `endian.h`, which on macos "it's complicated": https://stackoverflow.com/questions/20813028/endian-h-not-found-on-mac-osx

**Comment by DavidBuchanan314** [comment]
Ah, fun - the endian conversion only happens in one place, so I'll try to eliminate that dependency.

I'll fix the $ thing too - there's no warnings on my end and I thought it was fine, but I guess not!

**Comment by DavidBuchanan314** [comment]
I can't repro the uint64/size_t warning on my end, so I'm hoping that commit fixes it - there's some general jank with conversions between Py_ssize_t, size_t, and uint64_t which I'd like to clean up at some point

**Comment by DavidBuchanan314** [comment]
Did those commits silence the warnings for you?

**Comment by jamesmunns** [comment]
Looks good to me!

```
james@Jamess-MBP ➜  dag-cbrrr git:(main) pip3 install .      
Processing /Users/james/personal/satur-dag/vendor/dag-cbrrr
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: cbrrr
  Building wheel for cbrrr (pyproject.toml) ... done
  Created wheel for cbrrr: filename=cbrrr-0.0.1-cp311-cp311-macosx_13_0_arm64.whl size=24302 sha256=c7eeff0e23338595110b037bdec6ea9e58978b921fdfb52151a91f499ad4659e
  Stored in directory: /Users/james/Library/Caches/pip/wheels/0b/65/88/5a13ac15c4fd0e5231c6b5230269cc55e08c7b588cf94f0136
Successfully built cbrrr
Installing collected packages: cbrrr
  Attempting uninstall: cbrrr
    Found existing installation: cbrrr 0.0.1
    Uninstalling cbrrr-0.0.1:
      Successfully uninstalled cbrrr-0.0.1
Successfully installed cbrrr-0.0.1

[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python3.11 -m pip install --upgrade pip

james@Jamess-MBP ➜  dag-cbrrr git:(main) git lg | head -n 1                                                                                            
* 3ccdcca - (HEAD -> main, origin/main, origin/HEAD) Update README.md (2024-02-24T17:15:25+00:00) <David Buchanan>
```
